packages/commerce-sdk-react/src/provider.test.tsx

```diff
@@ -62,3 +62,33 @@ test(
         expect(screen.getByText(TEST_CONFIG.organizationId)).toBeInTheDocument()
     })
 )
+
+test(
+    'api clients optional config are passed properly',
+    withMocks(async () => {
+        const Component = () => {
+            const api = useCommerceApi()
+            return (
+                <ul>
+                    <li>{api?.shopperSearch?.clientConfig?.headers?.['correlation-id']}</li>
+                    <li>{api?.shopperSearch?.clientConfig?.fetchOptions?.timeout}</li>
+                </ul>
+            )
+        }
+        const optionalConfig = {
+            headers: {'correlation-id': '373a3f80-6bbb-4157-a617-63d27fb15769'},
+            fetchOptions: {
+                timeout: 50
+            }
+        }
+        renderWithProviders(
+            <Component />,
+            {},
+            {
+                optionalConfig
+            }
+        )
```

Comment:
Is it clear to the developers which provider this `optionalConfig` is for? I think without looking at the implementation, they would not know that it's actually for the `CommerceApiProvider`.

Reply:
How about something like this? This way is more future-proof. We can pass in props to any provider component.

```suggestion
        renderWithProviders(
            <Component />,
            {},
            {
                commerceApiProvider: {
                    headers: {'correlation-id': '373a3f80-6bbb-4157-a617-63d27fb15769'},
                    fetchOptions: {
                        timeout: 50
                    }
                }
            }
        )

```